metrics-core/src/main/java/com/codahale/metrics/LockFreeExponentiallyDecayingReservoir.java

```diff
@@ -52,6 +52,7 @@
 
     private final class State {
         private final long startTick;
+        // Count is updated after samples are successfully added to the map.
         private final AtomicLong count;
```

Comment:
I wonder if we're better off replacing the AtomicLong with AtomicLongFieldUpdater and a volatile long value. The initial increments have a chance to be slower due to noisy neighbor issues, however it will allow the fast path to avoid pointer jumping because the current value will live on the same cache line as the state object.

I'll try it and measure the results.

Reply:
No meaningful change, not worth the code churn I don't think.